src/6model/reprs/VMArray.c

```diff
@@ -599,6 +599,7 @@ static void pop(MVMThreadContext *tc, MVMSTable *st, MVMObject *root, void *data
         default:
             MVM_exception_throw_adhoc(tc, "MVMArray: Unhandled slot type");
     }
+    zero_slots(tc, body, body->start + body->elems, body->start + body->elems + 1, repr_data->slot_type);
```

Comment:
Would you mind introducing a const MVMuint64 slot = body->start + body->elems? Probably won't change anything for performance, but it would reduce code duplication for sure :)